### PR TITLE
session-service v1.0.0

### PR DESCRIPTION
**Pull request by mandawilson**
1) Added source and type, and saving data in different collections based on type.  Valid types are virtual_cohort and main_session.  Valid sources are anything at least 3 characters long.

2) Only unique JSON data is stored within a source and type.  If the user inserts data that is not unique, they get the ID of the previously stored data.  This is done with an index (defined in the code).

3) Custom query field=field&value=value.

4) Returns only id on create and blank response with 200 status on update, rather than returning the sessions.

I updated the documentation and unit tested all the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,161 +86,204 @@ session-service$ export set SERVER_PORT=8090; mvn package -Dpackaging.type=jar &
 
 ### Create
```
Comment (n1zea144) [inline README.md]: There is a tool called swagger that we've used to document the cbioportal webservice using a library called SpringFox.  We should integrate this at some point.  I can give you more information.

File: src/main/java/org/cbioportal/session_service/web/SessionServiceController.java [hunk 2]
```diff
@@ -56,55 +62,110 @@ public SessionServiceController(SessionRepository sessionRepository)
...
+        @PathVariable String type, 
+        @RequestParam(name="field") String field,
+        @RequestParam(name="value") String value)
     {
-        return sessionRepository.findAll();
+        List<Session> sessions = sessionRepository.findBySourceAndTypeAndQuery(source, type, field, value);
+        if (sessions.size() != 0) {
+            return sessions;
+        }
+        throw new SessionNotFoundException("field=" + field + "&value=" + value);
```
Comment (n1zea144) [inline src/main/java/org/cbioportal/session_service/web/SessionServiceController.java]: Take a look at the use of @ExceptionHandler here:

https://github.com/ersinciftci/mutation/blob/master/web/src/main/java/org/cbioportal/mutation/web/rest/MutationController.java

Comment (mandawilson) [inline src/main/java/org/cbioportal/session_service/web/SessionServiceController.java]: Let me know how this looks to you:

File: README.md (hunk 1 above)
Comment (mandawilson) [inline README.md]: I noticed this in someone's code.  If you could point me to the generated documentation and the source sometime that would be good.

File: src/main/java/org/cbioportal/session_service/web/SessionServiceController.java (hunk 2 above)
Comment (mandawilson) [inline src/main/java/org/cbioportal/session_service/web/SessionServiceController.java]: I cleaned things up a lot by adding a service package, among other things it helped with cleaning up the exception handling.

**Pull request merged**